horovod/common/controller.cc

```diff
@@ -273,12 +273,6 @@ ResponseList Controller::ComputeResponseList(bool this_process_requested_shutdow
 
         bool reduce = IncrementTensorCount(message, process_set.joined_size);
 
-        // For barrier request, if not ready to reduce, we add it back to tensor queue
```

Comment:
have you checked why this has been added in the first place?

Reply:
An equivalent statement was removed from one alternative control flow branch in an intermediate commit on the original barrier PR: https://github.com/horovod/horovod/pull/3139/commits/acf8aac7c7d6834a01dd10c45a3ba1c8ed5b3bef

I think the PR was remodeled quite a bit after its initial posting, so I assume this was an accidental remnant.